source/common/stream_info/stream_info_impl.h

```diff
@@ -371,6 +371,8 @@ struct StreamInfoImpl : public StreamInfo {
     filter_chain_name_ = info.filterChainName();
     attempt_count_ = info.attemptCount();
     upstream_bytes_meter_ = info.getUpstreamBytesMeter();
+    bytes_sent_ = info.bytesSent();
+    is_shadow_ = info.isShadow();
```

Comment:
Can we add some check or at least a comment in StreamInfo declaration about changing this place if new field is added?

Reply:
I am curious why  the asserts in `assertStreamInfoSize()` of StreamInfoImplTest are not triggered then?

Is this because you fixed the asserts when you added them initially but forgot to add the copies in setFrom?

Reply:
@danzh2010, as tyxia pointed out, `assertStreamInfoSize` is the intended mechanism for keeping this up to date if new fields are added.

@tyxia, that's right -- these fields were already there in streaminfo, but I didn't add them to this method initially.

Reply:
thanks for confirmation!